src/components/MovieList/MovieListComponent.js

```diff
@@ -1,13 +1,24 @@
 import React from 'react';
 import MovieCardComponent from '../MovieCard/MovieCardComponent';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {store, deleteMovieActionAsync, editMovieActionAsync} from '../../store/store';
 import './MovieListComponent.css';
 
 const MovieListComponent = (props) => {
   const { movies } = props;
-  if (movies.length === 0) {
-    throw new Error('no Movies found');
-  }
+
+  const editMovie = movie => {
+    store.dispatch(editMovieActionAsync(movie)).then(() => {
+      props.fetchMovies();
+    });
+  };
+
+  const deleteMovie =  movie => {
+    store.dispatch(deleteMovieActionAsync(movie)).then(() => {
```

Comment:
what i want to recommend you, let's try to move  store.dispatch method into EACH action inside))

and let's use _async and await_ for promise)